contracts/MarketCapSqrtController.sol

```diff
@@ -612,18 +612,22 @@ contract MarketCapSqrtController is MarketCapSortedTokenCategories {
 /* ==========  Internal Pool Utility Functions  ========== */
 
   /**
-   * @dev Estimate the total value of a pool by taking its first token's
-   * "virtual balance" (balance * (totalWeight/weight)) and multiplying
-   * by that token's average ether price from UniSwap.
+   * @dev Estimate the total value of a pool by taking the sum of
+   * TWAP values of the pool's balance in each token it has bound.
    */
-  function _estimatePoolValue(IIndexPool pool) internal view returns (uint144) {
-    (address token, uint256 value) = pool.extrapolatePoolValueFromToken();
-    return oracle.computeAverageEthForTokens(
-      token,
-      value,
+  function _estimatePoolValue(IIndexPool pool) internal view returns (uint256 totalValue) {
+    address[] memory tokens = pool.getCurrentTokens();
+    uint256 len = tokens.length;
+    uint256[] memory balances = new uint256[](len);
+    for (uint256 i; i < len; i++) balances[i] = IERC20(tokens[i]).balanceOf(address(pool));
```

Comment:
Safe to assume the actual pool value is tracking the actual balance accurately at any given time?

Reply:
Yeah, this is just as safe as using the balances to calculate mint outputs in the pool itself.